clang/lib/AST/Expr.cpp

```diff
@@ -2503,6 +2506,29 @@ Stmt *BlockExpr::getBody() {
 // Generic Expression Routines
 //===----------------------------------------------------------------------===//
 
+bool Expr::mayBranchOut() const {
```

Comment:
Computing this seems a little expensive in general. I wonder if we could track a bit on `FunctionDecl` that indicates whether it contains any branches out of statement expressions, and skip calling this if the enclosing function is not a coroutine and does not contain branches out of statement expressions.

Reply:
This is now no more needed. We switched to deleting instructions related to unemitted cleanups.